File: app/src/main/java/com/radionula/radionula/fragments/FavoritsFragment.java
package com.radionula.radionula.fragments;


import android.graphics.Typeface;
import android.os.Bundle;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;
import android.widget.TextView;

import com.radionula.radionula.model.NulaTrack;
import com.radionula.radionula.MyApp;
import com.radionula.radionula.PlaylistAdapter;
import com.radionula.radionula.R;

import java.util.List;

/**
 * A simple {@link Fragment} subclass.
 */
public class FavoritsFragment extends Fragment {

    ListView lvFavorites;
    PlaylistAdapter adapter;
    TextView tvHeader;

    public FavoritsFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view =  inflater.inflate(R.layout.fragment_favorits, container, false);

        lvFavorites = (ListView) view.findViewById(R.id.fragment_favorites_lvFavorites);
        tvHeader = (TextView)view.findViewById(R.id.fragment_favorites_tvHeader);

        List<NulaTrack> tracks = (List<NulaTrack>) MyApp.Companion.LoadUserFavorites(getActivity());
        if (tracks != null)
            setFavoritesList(tracks);

        Typeface artistFont = Typeface.createFromAsset(getActivity().getAssets(), "fonts/Roboto-Regular.ttf");
        tvHeader.setTypeface(artistFont);

        return view;
    }

    public void setFavoritesList(List<NulaTrack> tracks){
        adapter = new PlaylistAdapter(getActivity(), tracks, PlaylistAdapter.AdapterType.REMOVE);

        lvFavorites.setAdapter(adapter);


    }

}

File: app/src/main/java/com/radionula/radionula/favorits/FavoritsFragment.kt
package com.radionula.radionula.favorits

import android.graphics.Typeface
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import com.radionula.radionula.data.db.NulaDatabase
import com.radionula.radionula.databinding.FragmentFavoritsBinding
import com.radionula.radionula.model.NulaTrack
import kotlinx.coroutines.launch
import org.koin.android.ext.android.inject

class FavoritsFragment : Fragment() {
    private lateinit var binding: FragmentFavoritsBinding

    val nulaDatabase: NulaDatabase by inject()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        binding = FragmentFavoritsBinding.inflate(layoutInflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.fragmentFavoritesLvFavorites.layoutManager = LinearLayoutManager(activity)

        val artistFont = Typeface.createFromAsset(requireContext().assets, "fonts/Roboto-Regular.ttf")
        binding.fragmentFavoritesTvHeader.typeface = artistFont

        // Scoped to the view, so a query still in flight when the screen closes
        // is cancelled instead of landing on a dead binding.
        viewLifecycleOwner.lifecycleScope.launch {
            val tracks = nulaDatabase.selectAllTracks().toMutableList()
            binding.fragmentFavoritesLvFavorites.adapter =
                MyAdapter(tracks, ::removeFavorite, requireContext())
        }
    }

    private fun removeFavorite(track: NulaTrack) {
        viewLifecycleOwner.lifecycleScope.launch { nulaDatabase.removeTrack(track) }
    }
}
